### PR TITLE
Take into account CARGO_HOME environment variable in Crates Local Index

### PR DESCRIPTION
**Pull request by avrong**
changelog: Take into account the CARGO_HOME environment variable when using crates local index

### CODE REVIEW

File: toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt [hunk 1]
```diff
@@ -285,10 +284,20 @@ class CratesLocalIndexServiceImpl
...
-        // TODO: Determine how path to index is created
-        private const val CARGO_REGISTRY_INDEX_LOCATION: String = ".cargo/registry/index/github.com-1ecc6299db9ec823/.git/"
-        private const val CARGO_REGISTRY_INDEX_TAG_PRE_1_50 = "origin/master"
-        private const val CARGO_REGISTRY_INDEX_TAG = "origin/HEAD"
+        private val cargoHome: String
+            get() = System.getenv("CARGO_HOME")
+                ?: Paths.get(System.getProperty("user.home"), ".cargo/").toString()
+
+        // Currently for crates.io only
+        private val cargoRegistryIndexPath
```
Comment (ortem) [inline toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt:292]: Please specify the type explicitly

**Review by ortem — CHANGES_REQUESTED** [review]
Please handle the situation when `CARGO_HOME` is invalid (e.g. contains a non-existent path). Currently, `LocalFileSystem.getInstance().refreshAndFindFileByPath(cargoRegistryIndexRefsLocation)!!` call (after rebase onto master) leads to `java.lang.NullPointerException` without any user-friendly description

**Comment by avrong** [comment]
I changed how the `CARGO_HOME` environment variable is being retrieved as macOS doesn't bypass them from the shell, so `System.getEnv()` doesn't return any of those. To handle this, IDE calls shell environment on start to get envars using `printenv` and can return them using `EnviromentUtil`

**Review by ortem — APPROVED** [review]
LGTM

**Pull request merged**